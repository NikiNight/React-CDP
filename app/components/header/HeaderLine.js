import React from 'react';
import Logo from '../common/Logo';

class HeaderLine extends React.Component {
    render() {
        return (
            <div className="header-line">
                <Logo/>
                <button className="button button_reverse">Search</button>
            </div>
        )
    }
}

export default HeaderLine;