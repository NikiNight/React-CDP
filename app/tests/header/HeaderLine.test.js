import React from 'react';
import renderer from 'react-test-renderer';
import HeaderLine from './../../components/header/HeaderLine';
import {mount} from 'enzyme';

describe('Test HeaderLine component', () => {

  test('matches snapshot', () => {
    const tree = renderer.create(<HeaderLine isMovieSelected={true} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should be call backToSearch', () => {
    const backToSearch = jest.fn();
    const component = mount(<HeaderLine isMovieSelected={true} backToSearch={backToSearch}/>);
    component.find('.button').simulate('click');
    expect(backToSearch).toHaveBeenCalled();
  });

})