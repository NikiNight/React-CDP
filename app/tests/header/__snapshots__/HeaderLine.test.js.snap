// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test HeaderLine component matches snapshot 1`] = `
<div
  className="header-line"
>
  <a
    className="logo"
    href="#"
  >
    netflixroulette
  </a>
  <button
    className="button button_reverse"
    onClick={[Function]}
  >
    Search
  </button>
</div>
`;
